lib/swiss_village_directory/repository.rb

```diff
@@ -11,5 +11,23 @@ def initialize
         Village.new(row[0], row[1], row[2], row[3], row[4], row[5], row[6])
       end
     end
+
+    def find(find_params = {})
+      @villages.find_all { |v| filter_matched?(v, find_params) }
+    end
+
+    private
+
+    def filter_matched?(village, filter_params)
+      match = true
+      filter_params.each do |p|
+        begin
+          match = false if village.send(p.first) != p.last
+        rescue
+          match = false
+        end
+      end
+      match
+    end
```

Comment:
I don't like that error handling is used here. This should be possible in a fully functional way. Maybe better iterate over `self.instance_variables`.